### PR TITLE
Improved migration creation + Windows fix

### PR DESCRIPTION
**Pull request by r3code**
This PR incorporates all the features added by PR #312 to `migrate` tool, so it supersedes it. 
**What the difference?** 
1) Windows OS dir path issue fixed
2) `CreateCmd` tests fixed for Windows and Linux (`dir invalid` case especially)
3) each `*Cmd()` in `commands.go` now returns an error on fail and do not use log.fatalErr(err).
4) `migrate create` now prints out absolute path to created files.
5) Added comments requested by @dhui
6) `migrate` usage docs updated.

### CODE REVIEW

**Comment by r3code** [comment]
@dhui can you conduct a review?

File: internal/cli/commands_test.go [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
 	"path/filepath"
 	"strconv"
+	"strings"
 	"testing"
 	"time"
 
 	"github.com/stretchr/testify/suite"
 )
 
+var noErrorExpected error = nil
```
Comment (dhui) [inline internal/cli/commands_test.go]: There's no value in this variable. Just use `nil` directly for `expectedErr` in the tests

File: internal/cli/main.go [hunk 2]
```diff
@@ -20,6 +20,7 @@ const defaultTimeFormat = "20060102150405"
 // set main log
 var log = &Log{}
 
+// Main main function of a cli application. It is publiic for backwards compatibility with `cli` package
```
Comment (dhui) [inline internal/cli/main.go]: Make this `Main function` or `Main is the main function`

Comment (dhui) [reply internal/cli/main.go]: And `public` is misspelled

Comment (r3code) [reply internal/cli/main.go]: fixed in https://github.com/r3code/migrate/commit/f20abab955d9f8c80ad72c49b473bda834a486f5 and  https://github.com/r3code/migrate/commit/47c2c2c834d98d9f0eac914ab60098a81ae40819

File: internal/cli/commands_test.go (hunk 1 above)
Comment (r3code) [reply internal/cli/commands_test.go]: fixed in r3code@f20abab

**Pull request merged**